### DISCUSSION TITLE
Problem with solution for simple 3x3 matrix

### DISCUSSION THREAD

**Discussion by pierre321** [yalmip/YALMIP#1185] [Optimization modelling]
Dear community,

I want to solve a very simple linear problem for a 3x3 matrix. I am minimizing the sum of all 9 variables applying the following constraint:

x >= min,

with 

             min = [ 6 0 6;
                     0 6 6;
                     0 0 6].

Since I am minimizing the sum, the result should be the same as the min-matrix. Unfortunately I am getting the following result:

               x = [ 6 0 6;
                     0 6 6;
                     6 6 6].

Can anyone explain, why I am getting this result. I do not apply any further constraints. The code can be found below. Thank you in advance.

----------------------------------------------------------------------------------------------------------------------------------------------

yalmip('clear')

x = sdpvar(3,3);

min = [6,0,6;0,6,6;0,0,6];

constraints = x >= min;

objective = sum(sum((x)));

%% Solving

options = sdpsettings('verbose',1);
sol = optimize(constraints,objective,options);

**Comment by johanlofberg** [comment]
You are defining x to be symmetric
https://yalmip.github.io/tutorial/basics/

**Reply by pierre321** [reply]
Thank you! It works with adding 'full' while defining x
